sys/syz-extract/darwin.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2021 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package main
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/google/syzkaller/pkg/compiler"
+)
+
+type darwin struct{}
+
+func (*darwin) prepare(sourcedir string, build bool, arches []*Arch) error {
+	if sourcedir == "" {
```

Comment:
I see this is duplicated across all OSes. Please move this check into the caller.

Reply:
Done.